### PR TITLE
allow to add metadata directly on upload

### PR DESCRIPTION
**Pull request by chregu**

### CODE REVIEW

File: src/apis/sourceimages.js [hunk 1]
```diff
@@ -137,13 +137,19 @@ export default (state) => {
    *   .catch(function(err) {});
    * ```
    *
+   * With directly adding metadata:
+   *
+   * rokka.sourceimages.create('myorg', 'picture.png', file, {'meta_user': {'foo' => 'bar'}})
```
Comment (mweibel) [inline src/apis/sourceimages.js]: can you add triple backticks around the code?

File: src/apis/sourceimages.js [hunk 2]
```diff
@@ -171,8 +177,15 @@ export default (state) => {
       }
     })
     .then((data) => {
+      var formData = {}
```
Comment (mweibel) [inline src/apis/sourceimages.js]: please use `const`

Comment (mweibel) [reply src/apis/sourceimages.js]: actually that and the following lines could be simplified to:
```js
const formData = Object.assign({}, metadata);
```

can you try that out? I only tried it in my browser console.

Comment (chregu) [reply src/apis/sourceimages.js]: you sure? I also add a '[0]' to the keys of all entries (since rokka expects that here) in that forEach loop

Comment (mweibel) [reply src/apis/sourceimages.js]: ah, right.. somehow didn't look close enough 😅 
ok then that's fine if you change `var` to `const` (const, non-primitive (array, objects) variables can still be updated) and `const` has better scoping than `var`.

File: src/apis/sourceimages.js (hunk 1 above)
Comment (chregu) [reply src/apis/sourceimages.js]: added

**Comment by chregu** [comment]
Fixed that stuff and also added a test for the metadata on upload

**Pull request merged**